frameworks/spark/src/main/java/com/criteo/hadoop/garmadon/spark/listener/GarmadonSparkStorageStatusListener.java

```diff
@@ -0,0 +1,314 @@
+        liveRDDs.values().forEach(info -> info.distributions.remove(event.executorId()));
+    }
+
+    /**
+     * capture persistence mutation. Different types of blocs can be updated,, they either concern RDD, in which case we update both rdd and executor info,
+     * or just executors (stream, broadcast, etc...)
+     */
+    @Override
+    public void onBlockUpdated(SparkListenerBlockUpdated event) {
+        StorageLevel storageLevel = event.blockUpdatedInfo().storageLevel();
+        long diskDelta = event.blockUpdatedInfo().diskSize() * (storageLevel.useDisk() ? 1 : -1);
+        long memoryDelta = storageLevel.useOffHeap() ? 0 : event.blockUpdatedInfo().memSize() * (storageLevel.useMemory() ? 1 : -1);
+        long offHeapMemoryDelta = storageLevel.useOffHeap() ? event.blockUpdatedInfo().memSize() * (storageLevel.useMemory() ? 1 : -1) : 0;
+        BlockId blockId = event.blockUpdatedInfo().blockId();
+        String executorId = event.blockUpdatedInfo().blockManagerId().executorId();
+        if (blockId instanceof RDDBlockId) {
+            updateRDDMemoryStatus((RDDBlockId) blockId, storageLevel, executorId, diskDelta, memoryDelta, offHeapMemoryDelta);
+            updateExecutorRDDMemoryStatus(executorId, diskDelta, memoryDelta, offHeapMemoryDelta);
+        }
+        if (blockId instanceof StreamBlockId) {
+            updateExecutorStreamMemoryStatus(executorId, diskDelta, memoryDelta, offHeapMemoryDelta);
+        }
+        if (blockId instanceof BroadcastBlockId) {
+            updateExecutorBroadcastMemoryStatus(executorId, diskDelta, memoryDelta, offHeapMemoryDelta);
+        }
+    }
+
+    private void updateRDDMemoryStatus(RDDBlockId blockId, StorageLevel storageLevel, String executorId,
+                                       long diskDelta, long memoryDelta, long offHeapMemoryDelta) {
+        GarmadonRDDStorageInfo rddInfo = liveRDDs.get(blockId.rddId());
+        if (rddInfo != null) {
+            //Global info
+            rddInfo.offHeapMemoryUsed = addDelta(rddInfo.offHeapMemoryUsed, offHeapMemoryDelta);
+            rddInfo.memoryUsed = addDelta(rddInfo.memoryUsed, memoryDelta);
+            rddInfo.diskUsed = addDelta(rddInfo.diskUsed, diskDelta);
+
+            //partition update, one partition == one block
+            RDDPartition partition = rddInfo.partitions.computeIfAbsent(blockId.name(), key -> new RDDPartition());
+
+            //compute block delta as number of block is used to remove or not a distribution from an RDD
+            int blockDelta = 0;
+            if (storageLevel.isValid()) {
+                if (!partition.executors.contains(executorId)) {
+                    blockDelta += 1;
+                    partition.executors.add(executorId);
+                }
+            } else {
+                blockDelta -= 1;
+                partition.executors.remove(executorId);
+            }
+
+            //remove a partition that leaves on no executor
```

Comment:
Shouldn't onExecutorRemoved get called in this case? In any case, shouldn't we call the same method (or send an event similarly to what you're doing with sendRDDStorageStatusEvent)?

Reply:
Which in the end helps to remove the correct amount of memory from an executor when an RDD is unpersisted

Reply:
Output was you could fix the typo: "leaves" -> "lives"